rclpy/rclpy/node.py

```diff
@@ -1949,3 +1950,25 @@ def get_subscriptions_info_by_topic(
             topic_name,
             no_mangle,
             _rclpy.rclpy_get_subscriptions_info_by_topic)
+
+    def wait_for_node(
+        self,
+        node_name: str,
+        timeout: float
+    ) -> bool:
+        """
+        Wait until node name is present in the system or timeout.
+
+        :param node_name: name of the node to wait for.
+        :param timeout: seconds to wait for the node to be present. If negative, the function
+                         won't timeout.
+        :return: True if the node was found, False if timeout.
+        """
+        start = time.time()
+        flag = False
+        # TODO refactor this implementation when we can react to guard condition events, or replace
+        # it entirely with an implementation in rcl. see https://github.com/ros2/rclpy/issues/929
+        while time.time() - start < timeout and not flag:
+            flag = node_name in self.get_node_names()
```

Comment:
The node name is not an unique identifier for a node, `get_node_names_and_namespaces()` should be used instead.
Having a `get_fully_qualified_node_names()` would be useful as well.

Reply:
Good catch. I've updated this PR to account for fully qualified node names.